test/parallel/test-assert.js

```diff
@@ -646,6 +646,34 @@ try {
 }
 
 {
+  let threw = false;
+  const rangeError = new RangeError('my range');
+
+  // verify custom errors
+  try {
+    assert.strictEqual(1, 2, rangeError);
+  } catch (e) {
+    assert.strictEqual(e, rangeError);
+    threw = true;
+    assert.ok(e instanceof RangeError, 'Incorrect error type thrown');
+  }
+  assert.ok(threw);
+  threw = false;
```

Comment:
It would actually be nice to use `common.expectsError` here. The test would be smaller in that case. You can check some other use cases as a reference.

Reply:
Thank you for the suggestion. I tried `expectsError`, but for this simple case it didn't make the code any cleaner/shorter/simpler. I'm keeping it as is for now.